src/bootstrap/bootstrap.py

```diff
@@ -16,22 +16,39 @@
 import subprocess
 import sys
 import tarfile
+import tempfile
+
 
 def get(url, path, verbose=False):
-    print("downloading " + url)
     sha_url = url + ".sha256"
-    sha_path = path + ".sha256"
-    for _url, _path in ((url, path), (sha_url, sha_path)):
-        # see http://serverfault.com/questions/301128/how-to-download
-        if sys.platform == 'win32':
-            run(["PowerShell.exe", "/nologo", "-Command",
-                 "(New-Object System.Net.WebClient)"
-                 ".DownloadFile('{}', '{}')".format(_url, _path)],
-                verbose=verbose)
-        else:
-            run(["curl", "-o", _path, _url], verbose=verbose)
-    print("verifying " + path)
-    with open(path, "rb") as f:
+    temp_file = tempfile.NamedTemporaryFile(delete=False)
+    temp_path = temp_file.name
+    sha_file = tempfile.NamedTemporaryFile(suffix=".sha256", delete=True)
+    sha_path = sha_file.name
+    download(sha_path, sha_url, verbose)
+    download(temp_path, url, verbose)
+    verify(sha_path, temp_path, verbose)
+    sha_file.close()
+    print("moving " + temp_path + " to " + path)
+    shutil.move(temp_path, path)
+    temp_file.close()
```

Comment:
I find it strange that the file is closed only after attempting to move it. This may cause problems on Windows (not really sure tho). Since this tempfile won't get deleted on closing, why not close it earlier?